examples/nucleo_g474re/imu_lsm6dso/main.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2023, Raphael Lehmann
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <modm/board.hpp>
+#include <modm/platform/spi/spi_master_1.hpp>
+#include <modm/driver/inertial/lsm6dso.hpp>
+#include <optional>
+
+using namespace Board;
+
+using SpiMaster = SpiMaster1;
+using Mosi = GpioA7;
+using Miso = GpioA6;
+using Sck = GpioA5;
+using Cs = GpioC5;
+using Int1 = GpioC4;
+using ImuVdd = GpioA10;
+
+using Lsm6dso = modm::Lsm6dso<modm::Lsm6TransportSpi<SpiMaster, Cs>>;
+
+Lsm6dso imu{};
+
+int
+main()
+{
+	Board::initialize();
+	Leds::setOutput();
+
+	MODM_LOG_INFO << "LSM6DSO IMU demo on Nucleo-G474RE\n" << modm::endl;
+
+	SpiMaster::initialize<Board::SystemClock, 5300_kHz>();
+	SpiMaster::connect<Sck::Sck, Mosi::Mosi, Miso::Miso>();
+	Int1::setInput(Gpio::InputType::PullUp);
+
+	// The imu behaves weird when the pins are temporarily unconnected (e.g.
+	// while changing jumper wires) or after reprogramming the nucleo board,
+	// and a software reset does not work.
+	// Therefore the IMU can be powered through a GPIO pin (Vdd) which we
+	// switch on and off to reboot it completely.
+	ImuVdd::reset();
+	ImuVdd::setOutput();
+	modm::delay(10ms);
+	// Power up imu
+	ImuVdd::set();
+
+	// Give the sensor some time to boot...
+	modm::delay(10ms);
+
+	MODM_LOG_INFO << "Resetting and rebooting imu (BOOT | SW_RESET)..." << modm::endl;
+	// Write bits BOOT | SW_RESET (and IF_INC, because it default to 1) to reboot imu
+	if(!RF_CALL_BLOCKING(imu.writeRegister(modm::lsm6dso::Register::CTRL3_C, 0b1000'0101))) {
+		MODM_LOG_INFO.printf(" ... failed.");
+		Board::Leds::set();
+	}
```

Comment:
Shouldn't this be a named function in the driver or done in `initialize()` if the reboot is required?

Reply:
There are no intructions in STs documentation about how to use this bit or how to reset/reboot the IMU.
Also removing this code block does not change the behavior of IMU in this example, mainly because the IMU is resetted by removing the power supply just a few lines above.

I kept the lines in this example because in some (future) applications someone might need to reset the IMU configuration...

Reply:
Ok, thanks for the clarification.

Reply:
So you are okay with keeping this as it is?

Reply:
Yes